components/CategoryFilter.js

```diff
@@ -0,0 +1,26 @@
+import useSWR from "swr";
+
+export default function CategoryFilter({ onSelectCategory }) {
+  const { data: categories, isLoading, error } = useSWR("api/categories");
+
+  if (isLoading) {
+    return <p>Loading...</p>;
+  }
+  if (error) {
+    return <p>We could not load the categories...</p>;
+  }
+
+  return (
+    <div>
+      {categories.map((category) => (
+        <button
+          key={category._id}
+          onClick={() => onSelectCategory(category.category)}
+        >
+          <span>{category.emoji}</span>
+          <span>{category.category}</span>
+        </button>
+      ))}
+    </div>
```

Comment:
```suggestion
    </>
```
The `<div>` is not really needed here. You can replace it with a fragment, which won’t add an extra element to the DOM.